packages/netlify-plugin-search-index/index.js

```diff
@@ -0,0 +1,108 @@
+const { tmpdir } = require('os')
+const path = require('path')
+const fs = require('fs')
+const { promisify } = require('util')
+const {
+  env: { DEPLOY_ID },
+} = require('process')
+
+const cpy = require('cpy')
+const { zipFunctions } = require('@netlify/zip-it-and-ship-it')
+const htmlToText = require('html-to-text')
+
+const isNetlifyCI = require('../build/src/utils/is-netlify-ci') // eslint-disable-line node/no-unpublished-require
+
+function netlifyPluginSearchIndex(pluginConfig) {
+  const searchIndexFolder = pluginConfig.searchIndexFolder || 'searchIndex'
+  return {
+    name: '@netlify/plugin-search-index',
+    // scopes: ['listSites'],
+
+    async postBuild(opts) {
+      const { config } = opts
+      const { build } = config
+      const { BUILD_DIR } = opts.constants // where we start from
+
+      if (typeof BUILD_DIR === 'undefined') {
+        throw new Error('must specify publish dir in netlify config [build] section')
+      }
+
+      const buildFolderPath = path.resolve(BUILD_DIR)
+      let newManifest = []
+      newManifest = await walk(buildFolderPath)
+      newManifest = newManifest.filter(x => x.endsWith('.html'))
+      let searchIndex = {}
+      let customOpts = {
+        // TODO: expose this
+        // https://www.npmjs.com/package/html-to-text#options
+        wordwrap: 130,
+      }
+      const readfile = promisify(fs.readFile)
+      await Promise.all(
+        newManifest.map(htmlFilePath => {
+          return readfile(htmlFilePath, 'utf8').then(htmlFileContent => {
+            const text = htmlToText.fromString(htmlFileContent, customOpts)
+            const indexPath = path.relative(buildFolderPath, htmlFilePath)
+            searchIndex[`/${indexPath}`] = text
+          })
+        }),
+      )
+      let searchIndexPath = path.join(buildFolderPath, searchIndexFolder, 'searchIndex.json')
+      if (fs.existsSync(searchIndexPath)) {
```

Comment:
Should we use async I/O instead of sync? For example [`path-exists`](https://github.com/sindresorhus/path-exists) or using `util.promisify`.